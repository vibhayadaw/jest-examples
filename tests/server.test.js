const request = require('supertest');
const app = require('../src/server');

describe('the server API', () => {

    it('returns all users', async () => {
        const response = await request(app).get('/users');
        expect(response.body).toMatchSnapshot();
    });
    it('returns one user', async () => {
        const response = await request(app).get('/user/1');
        expect(response.body).toMatchSnapshot();
    });
});