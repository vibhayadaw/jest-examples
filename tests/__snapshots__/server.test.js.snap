// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the server API returns all users 1`] = `
Object {
  "1": Object {
    "age": 29,
    "id": 1,
    "name": "Sam",
    "sex": "male",
  },
  "2": Object {
    "age": 10,
    "id": 2,
    "name": "Taylor",
    "sex": "male",
  },
  "3": Object {
    "age": 11,
    "id": 3,
    "name": "Cleo",
    "sex": "male",
  },
}
`;

exports[`the server API returns one user 1`] = `
Object {
  "age": 29,
  "id": 1,
  "name": "Sam",
  "sex": "male",
}
`;
